exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -55,41 +56,223 @@ func (r *AzureManagedMachinePool) Default(client client.Client) {
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (r *AzureManagedMachinePool) ValidateCreate(client client.Client) error {
 	azuremanagedmachinepoollog.Info("validate create", "name", r.Name)
+	var allErrs field.ErrorList
+
+	// If AutoScaling is enabled, both MinCount and MaxCount should be set
+	if r.Spec.AutoScaling != nil && (r.Spec.AutoScaling.MaxCount == nil || r.Spec.AutoScaling.MinCount == nil || (*r.Spec.AutoScaling.MaxCount < *r.Spec.AutoScaling.MinCount)) {
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("Spec", "AutoScaling"),
+				r.Spec.AutoScaling,
+				"MaxCount and MinCount must be set and MaxCount must be greater or equal to MinCount when AutoScaling is enabled"))
+	}
+
+	if len(allErrs) != 0 {
+		return apierrors.NewInvalid(GroupVersion.WithKind("AzureManagedMachinePool").GroupKind(), r.Name, allErrs)
+	}
 	return nil
 }
 
+// ValidateUpdateHelper is a helper function to reduce cyclomatic complexity.
+func ValidateUpdateHelper(old AzureManagedMachinePoolSpec, r AzureManagedMachinePoolSpec) field.ErrorList {
+	var updateErrs field.ErrorList
+	if r.SKU != old.SKU {
+		updateErrs = append(updateErrs,
+			field.Invalid(
+				field.NewPath("Spec", "SKU"),
+				r.SKU,
+				"field is immutable"))
+	}
+
+	if old.OSDiskSizeGB != nil {
+		// Prevent OSDiskSizeGB modification if it was already set to some value
+		if r.OSDiskSizeGB == nil {
+			// unsetting the field is not allowed
+			updateErrs = append(updateErrs,
+				field.Invalid(
+					field.NewPath("Spec", "OSDiskSizeGB"),
+					r.OSDiskSizeGB,
+					"field is immutable, unsetting is not allowed"))
+		} else if *r.OSDiskSizeGB != *old.OSDiskSizeGB {
+			// changing the field is not allowed
+			updateErrs = append(updateErrs,
+				field.Invalid(
+					field.NewPath("Spec", "OSDiskSizeGB"),
+					*r.OSDiskSizeGB,
+					"field is immutable"))
+		}
+	}
+	if !reflect.DeepEqual(r.NodeTaints, old.NodeTaints) {
+		updateErrs = append(updateErrs,
+			field.Invalid(
+				field.NewPath("Spec", "NodeTaints"),
+				r.NodeTaints,
+				"field is immutable"))
+	}
```

Comment:
afaik taints on a kubernetes node are mutable, please correct me if i am wrong !!

Reply:
AKS limitation today: https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools#setting-nodepool-taints (see the box a bit below that link)

Reply:
Thanks 😊